### ISSUE TITLE
Stock items not listed + System Information / About InvenTree buttons not working

### ISSUE THREAD

**Issue by JustusRijke** [inventree/InvenTree#8085]
### Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find a similar issue

### Describe the bug*

After updating to 0.16.1, after a few days I noticed that i can no longer view stock items (url `/stock`).
The list is empty, but it knows it has many stock items. See screenshot.

![2024-09-05 20_38_16-InvenTree-PT _ Stock — Mozilla Firefox](https://github.com/user-attachments/assets/2f848e9f-69d5-43b9-a656-6e6aa32e92ad)

In addition, the System Information / About InvenTree buttons are no longer working (no effect after click), unless I am in the main menu (i.e. after clicking the InvenTree logo in the top left corner).

Unable to reproduce this on the demo site.

Possibly related to #8078 ?...

### Steps to Reproduce

go to "Stock" menu, click on System Information or About InvenTree

### Expected behaviour

show stock, buttons work

### Deployment Method

- [ ] Docker
- [ ] Package
- [X] Bare metal
- [ ] Other - added info in Steps to Reproduce

### Version Information

# Version Information:
InvenTree-Version: 0.16.1
Django Version: 4.2.15

Database: postgresql
Debug-Mode: False
Deployed using Docker: False
Platform: Linux-5.10.0-32-amd64-x86_64-with-glibc2.31
Installer: None

Active plugins: [{'name': 'InvenTreeBarcode', 'slug': 'inventreebarcode', 'version': '2.1.0'}, {'name': 'InvenTreeCoreNotificationsPlugin', 'slug': 'inventreecorenotificationsplugin', 'version': '1.0.0'}, {'name': 'InvenTreeCurrencyExchange', 'slug': 'inventreecurrencyexchange', 'version': '1.0.0'}, {'name': 'InvenTreeLabel', 'slug': 'inventreelabel', 'version': '1.1.0'}, {'name': 'InvenTreeLabelMachine', 'slug': 'inventreelabelmachine', 'version': '1.0.0'}, {'name': 'InvenTreeLabelSheet', 'slug': 'inventreelabelsheet', 'version': '1.0.0'}, {'name': 'DigiKeyPlugin', 'slug': 'digikeyplugin', 'version': '1.0.0'}, {'name': 'LCSCPlugin', 'slug': 'lcscplugin', 'version': '1.0.0'}, {'name': 'MouserPlugin', 'slug': 'mouserplugin', 'version': '1.0.0'}, {'name': 'TMEPlugin', 'slug': 'tmeplugin', 'version': '1.0.0'}]

### Please verify if you can reproduce this bug on the demo site.

- [ ] I can reproduce this bug on the demo site.

**Comment by SchrodingersGat** [comment]
@JustusRijke have you tried refreshing the cache on your browser?

https://docs.inventree.org/en/latest/faq/#feature-x-does-not-work-after-update

**Comment by Data-Sorted** [comment]
Having the same Issue after just updating.

Browser cache refresh made no difference, and having the same issue when using the Android App.

Parts show up on the index page and can be found through the search bar.

When on the Stock or Part page the "Scan Barcode", "System Information", and "About InvenTree" buttons dont function.
However they seem to work on the other pages. (I only tried "/company/suppliers" and "/company/customers") 

Looking at the server logs, it doesn't look like the server is getting the GET requests for /stats or /about when on the mentioned problem pages.



### API Version: 232
## Version Information:
InvenTree-Version: 0.16.1
Django Version: 4.2.15
Commit Hash: 4c15e5c
Commit Date: 2024-08-30

Database: postgresql
Debug-Mode: False
Deployed using Docker: True
Platform: Linux-6.1.0-23-amd64-x86_64-with
Installer: DOC

Active plugins: [{'name': 'InvenTreeBarcode', 'slug': 'inventreebarcode', 'version': '2.1.0'}, {'name': 'InvenTreeCoreNotificationsPlugin', 'slug': 'inventreecorenotificationsplugin', 'version': '1.0.0'}, {'name': 'InvenTreeCurrencyExchange', 'slug': 'inventreecurrencyexchange', 'version': '1.0.0'}, {'name': 'InvenTreeLabel', 'slug': 'inventreelabel', 'version': '1.1.0'}, {'name': 'InvenTreeLabelMachine', 'slug': 'inventreelabelmachine', 'version': '1.0.0'}, {'name': 'InvenTreeLabelSheet', 'slug': 'inventreelabelsheet', 'version': '1.0.0'}, {'name': 'DigiKeyPlugin', 'slug': 'digikeyplugin', 'version': '1.0.0'}, {'name': 'LCSCPlugin', 'slug': 'lcscplugin', 'version': '1.0.0'}, {'name': 'MouserPlugin', 'slug': 'mouserplugin', 'version': '1.0.0'}, {'name': 'TMEPlugin', 'slug': 'tmeplugin', 'version': '1.0.0'}]

**Comment by SchrodingersGat** [comment]
When you click on these buttons, etc, can you look in your browser console (in the developer tab) to see if there are any error or warning messages?

**Comment by Data-Sorted** [comment]
Pressing the buttons, nothing appears in the console.

On load, I get
`- Uncaught ReferenceError: loadApiIconPacks is not defined
    <anonymous> https://**URL**/stock/#:716
    jQuery 13
        l
        c
        setTimeout handler*Deferred/then/a/<
        u
        fireWith
        fire
        u
        fireWith
        ready
        _
        EventListener.handleEvent*
        <anonymous>
        <anonymous>
**URL**:716:5`

`Source map error: request failed with status 404
Resource URL: https://**URL**.com/static/script/chartjs-adapter-moment.js
Source Map URL: chartjs-adapter-moment.min.js.map
`

`Source map error: NetworkError when attempting to fetch resource.
Resource URL: https://**URL**.com/static/script/moment.js
Source Map URL: /sm/ac6ca6e51d8435fbb492538442f7060550b81bb0c2505d4048856d957213e3a8.map
`
The 2 "Source map error" warnings both show up on the "/customers/" page, but the Reference Error doesn't

**Comment by SchrodingersGat** [comment]
@Data-Sorted I have tried to reproduce this on a server running `0.16.1` - but without success.

Can you try running `invoke static` first, to check if maybe the static js files are being correctly collected?

**Comment by Data-Sorted** [comment]
Doesn't look like it threw any warnings related to the static filescl

/home/inventree # invoke static
Python version 3.11.9 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Rendered all files in /home/inventree/src/backend/InvenTree/templates/js/translated
Collecting static files...
Python version 3.11.9 - /usr/local/bin/python3
Python version 3.11.9 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")

**Comment by Data-Sorted** [comment]
Still pretty new to the repo but I noticed that "loadApiIconPacks" is related to localization??

Anyway when I changed the language to Deutsch the error went away and the problems dissapeared. Only seems to be an Issue when the language is set to English

**Comment by SchrodingersGat** [comment]
> Anyway when I changed the language to Deutsch the error went away and the problems dissapeared. Only seems to be an Issue when the language is set to English

Even stranger. Does the problem return if you switch back to english?

**Comment by Data-Sorted** [comment]
Yup, goes right back to its previous state.

**Comment by SchrodingersGat** [comment]
Ok, then it might be worth looking in the "network" tab in the developer console, and filtering requests by "javascript". Then, do a hard page refresh and see if the browser is failing to load a particular javascript file...

**Comment by Data-Sorted** [comment]
No failed requests.

On the /stock/ page in English there's no request for en.helpers.js. However, de.helpers.js is present when the language is switched to Deutsch.

On the /build/ page both de.helpers.js and en.helpers.js show up in their respective languages

**Comment by Data-Sorted** [comment]
After messing with the language setting a clearing the cache a few more times. It seems to have fixed itself??
No idea what changed, and I haven't been able to replicate it again yet.

**Comment by stavismed** [comment]
I have the same problem, but switching the language (englisch / deutsch) and clearing browser cache does not help.
The left navigation does not show up on stocks or parts.

**Comment by JustusRijke** [comment]
`invoke update` fixed it for me as well!
100% sure I ran `invoke update` after updating to 0.16.1 tho...

**Comment by SchrodingersGat** [comment]
Glad you both got it working :)

**Comment by nlsrchtr** [comment]
Hi @SchrodingersGat,

i was facing the same situation and was not able to resolve the situation fully. After diving into this deeper, I found that the NGINX config you are providing are setting a 30 day cache via the NGINX headers to the browser and potentially other caching systems in-between.

In my case I'm running InvenTree behind Cloudflare, which is also caching the files on it's side. So forcing my browser to clear the cache doesn't solve the problem, since Cloudflare would also serve the file for the next 30 days from it's cache. Since the method is defined in the languages files, I was just adding a fake parameter to the URL (`/static/i18n/de.helpers.js?test=1`) and I saw the `loadApiIconPacks` method appear.

From this investigation, I would highly suggest to lower / remove the caching time from the NGINX configuration or event better, to work with "fingerprinted" asset urls like `/static/i18n/de.helpers.js?($timestamp_of_creation_time_of_js_file)`.

I was only able to work around this, by switching to a different language, since the German JS file is still served by the cache of Cloudflare.

Regards,

**Comment by matmair** [comment]
This problem only affects the old interface - which is slated for removal. The new interface contains hashes in the filenames and is optimized to be cached so I do not think it makes sense to tackle this as it is very uncommon.

**Comment by nlsrchtr** [comment]
Hi @matmair,

since the new UI is not set as default, I was under the impression, that it's not useable on a daily base. Since the "old UI" is still the default, I think most people are using it. I don't know how much effort it is to add the hashes to the filenames, but it would help users to not run into the same issue again.

_Thanks for your great work on this very useful project!_

**Comment by matmair** [comment]
@nlsrchtr it is a lot of effort and we are 2.5 devs on this project, this will not be addressed from the core team.

**Comment by nikolai2111** [comment]
> In my case I'm running InvenTree behind Cloudflare

@nlsrchtr you are using the inventree behind a Cloudflare tunnel? How have you setup the ` .env` files. I'm not able to connect my inventree to the tunnel since inventree updated from ngnix to caddy.

This is my current ` .env` but I'm only available to connect from `192.168.123.123`.
```
INVENTREE_SITE_URL="http://192.168.123.123"
INVENTREE_TRUSTED_ORIGINS=["http://192.168.212.50","https://inventree.example.com"]
```

I Cloudflare I have configured the tunnel like shown below:
Public hostname: inventree.example.com
Service: http://192.168.123.123

Do you have any suggestion for me?

**Comment by nlsrchtr** [comment]
Hi @nikolai2111,

I'm using the NGINX setup with [this](https://github.com/inventree/InvenTree/blob/master/contrib/container/nginx.conf) config. Since this is used for serving the static files only, that works for me.

There is only the `INVENTREE_SITE_URL` environment variable available, related to those settings. I'm not using `INVENTREE_TRUSTED_ORIGINS`.

Hope this helps.
